Wikipedia/View Controllers/Nearby/NearbyViewController.m

```diff
@@ -122,12 +113,50 @@ - (BOOL)prefersStatusBarHidden {
     return NO;
 }
 
-- (void)viewWillDisappear:(BOOL)animated {
-    [self.locationManager stopUpdatingLocation];
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+
+    self.skipStartUpdates = NO;
+    // Only begin updates if by the time the startUpdates selector below is performed
+    // self.skipStartUpdates is still NO - lets us add the nearby view controller
+    // somewhere and immediately cover it up without it starting to update locations
+    // and headings.
+    [self performSelector:@selector(startUpdates) withObject:nil afterDelay:0.5];
```

Comment:
This is a bit of a hack, but I kind of prefer this solution as it means you can still just add a nearby view controller and it *just works* without having to conditionally call some public "start working" method. And you can add a nearby view controller and cover it immediately without it throwing up a "app wants location data access" permissions popup over the top of the covering view controller's view.